doc/source/user_guide/tutorials/data_sources.rst

```diff
@@ -0,0 +1,785 @@
+
+This tutorial describes the behavior of the ``DataSources`` class
+for all files formats accepted by PyDPF-Core:
+
+.. include:: ../../substitution_solvers.rst
+
+.. _ref_list_extensions_solvers:
+
+Solver output formats supported by DPF
+++++++++++++++++++++++++++++++++++++++
+
+   - MAPDL files: |MAPDL_files|
+   - LS-DYNA files: |LS-DYNA_files|
+   - Fluent files: |Fluent_files1| and |Fluent_files2|
+   - CFX files: |CFX_files1| and |CFX_files2|
+
+This tutoriam will present a general code, with generic files, and examples
+with results files available with the PyDPF-Core package.
+
+DataSources creation
+--------------------
+
+**a) Class** :class:`DataSources <ansys.dpf.core.data_sources.DataSources>`
+When dealing with a single file of a format listed above,
+instantiate a ``DataSources`` directly with the
+file path as an argument :
+
+.. tab-set::
+
+    .. tab-item:: Generic code
+
+        .. code-block:: python
+
+            from ansys.dpf import core as dpf
+
+            # Create a DataSources object for a single file
+            my_data_sources_a = dpf.DataSources(result_path=r'file.extension')
+
+    .. tab-item:: Example
+
+        .. code-block:: python
+
+            from ansys.dpf import core as dpf
+            from ansys.dpf.core import examples
+
+            # Create a DataSources object for a single file
+            my_data_sources_a = dpf.DataSources(result_path=examples.find_simple_bar())
+
+Generate a raw string (by putting the letter 'r' before
+the file path string) in order to ensure the file path is read correctly.
+
+**b) Function** :func:`set_result_file_path() <ansys.dpf.core.data_sources.DataSources.set_result_file_path>`
+
+If the extension key is not in the list above (For example: a file.binout can
+sometimes be named as 'file.binout0000') but you know what the extension key is,
+the DataSources object can be created by:
+
+.. tab-set::
+
+    .. tab-item:: Generic code
+
+        .. code-block:: python
+
+            from ansys.dpf import core as dpf
+
+            # Create the DataSources object
+            my_data_sources_b = dpf.DataSources()
+            # Define the path where the main result data can be found
+            my_data_sources_b.set_result_file_path(filepath=r'file.extension1234', key='extension')
+
+    .. tab-item:: Example
+
+        .. code-block:: python
+
+            from ansys.dpf import core as dpf
+            from ansys.dpf.core import examples
+
+            # Create the DataSources object
+            my_data_sources_b = dpf.DataSources()
+            # Define the path where the main result data can be found
+            path = examples.find_simple_bar()
+            'C:/Users/user/AppData/local/temp/ASimpleBar.rst'
+            my_data_sources_b.set_result_file_path(filepath=r'C:/Users/user/AppData/local/temp/ASimpleBar.rst', key='rst')
+
+**c) Function** :func:`guess_result_key() <ansys.dpf.core.data_sources.DataSources.guess_result_key>`
```

Comment:
Hey @jorgepiloto we are seeing annoying errors from Vale which does not exclude links to refs for functions. Do you know how to easily to solve this? Thanks!

Reply:
Let me double check the configuration

Reply:
I am not sure how I was able to accidentally push the commit to `master`. This branch is protected and forcing a pull-request. Moreover, nobody can overcome previous restriction. The commit is this one: https://github.com/ansys/pydpf-core/commit/ca11a17499f35f6a7d8f6be885a27b393aad0589

Reply:
Just cherry-picked in https://github.com/ansys/pydpf-core/pull/1774/commits/912790994a168749a4130134ac667bcb6ca90ae9